sustaingym/envs/building/stochastic_uncontrollable_generator.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+
+import numpy as np
+import pandas as pd
+import scipy
+
+
+class StochasticUncontrollableGenerator():
+    def __init__(self, num_episodes: int | None = None):
+        """
+        Initializes a generator class for the uncontrollable features in BuildingEnv.
+
+        Args:
+            num_episodes: The number of episodes in the provided observation data. If
+                given, instance will treat each row in the observation data as a separate
+                raw dataset to split into seasons and fit distributions to. If None,
+                will use the provided raw data as a single episode.
+        """
+        self.episodes = num_episodes
+
+        self.observations = [[]]
+
+        self.summer_observations = []
+        self.winter_observations = []
+
+        self.summer_dists = None
+        self.winter_dists = None
+        
+        self.block_size = 0
+
+    def split_observations_into_seasons(
+        self, observation_data: np.ndarray
+    ) -> (np.ndarray, np.ndarray):
```

Comment:
For typing annotations of return types, if it's a 2-tuple, then you should write
```python
def f(x: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
   # code
```